### PR TITLE
support listen verb support bidirectionalAudioSampleRate

### PR DESCRIPTION
**Pull request by xquanluu**
https://github.com/jambonz/freeswitch-modules/issues/25

### CODE REVIEW

File: lib/tasks/listen.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ class TaskListen extends Task {
       mixType: this.mixType,
       sampling: this.sampleRate,
       ...(this._bugname && {bugname: this._bugname}),
-      metadata
+      metadata,
+      bidirectionalAudio: this.bidirectionalAudio || {}
```
Comment (davehorton) [inline lib/tasks/listen.js:137]: disableBidirectionalAudio is there for legacy purposes, so we have to keep it.  But I think in this listen.js code we should immediately set `bidirectionalAudio.enabled` here and then in the rest of the code refer only to that, ie.right after the line above do something like
```
if ('enabled' not in this.bidirectionalAudio) {
   this.bidirectionalAudio.enabled = this. disableBidirectionalAudio === true ? false : true;
}
```

**Review by davehorton — CHANGES_REQUESTED** [review]

**Comment by davehorton** [comment]
listen tests are failing

File: lib/tasks/listen.js (hunk 1 above)
Comment (davehorton) [reply lib/tasks/listen.js:142]: are you going to address this?

Comment (xquanluu) [reply lib/tasks/listen.js:142]: Yes doing it now

Comment (xquanluu) [reply lib/tasks/listen.js:150]: fixed

**Comment by davehorton** [comment]
I think we need to explicitly handle the case where the feature has been updated to have the bidirectional streaming feature but the freeswitch has not.  Could you make sure this results in the feature-server logging an error message and ending the listen task, proceeding to the next task

**Comment by xquanluu** [comment]
> I think we need to explicitly handle the case where the feature has been updated to have the bidirectional streaming feature but the freeswitch has not. Could you make sure this results in the feature-server logging an error message and ending the listen task, proceeding to the next task

The issue is, the forkAudioStart will not be failed on feature server due to freeswitch still be able to open ws to destination as normal, but bugname or metadata could be wrong due to we added bidirectionalAudio params.

**Comment by davehorton** [comment]
can you rebase to main?

**Comment by xquanluu** [comment]
rebased

**Comment by davehorton** [comment]
I updated the freeswitch image, test passing now, can you rebase again @xquanluu ?

**Review by davehorton — APPROVED** [review]

**Pull request merged**